### PR TITLE
Red: Super late submission for chapter 02. Apologies all

### PR DESCRIPTION
**Pull request by utopia-2**

### CODE REVIEW

File: src/red/java/com/lambdaherding/edi/red/ch02/Calculator.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	}
+
+	private JPanel makeButtons() {
+		JPanel pan = new JPanel();
+		pan.setLayout( new GridLayout( 4, 5 ) );
+
+		pan.add( inputButton( "7" ) );
+		pan.add( inputButton( "8" ) );
+		pan.add( inputButton( "9" ) );
+		pan.add( binaryActionButton( "/", (a, b) -> 1 / a * b ) );
```
Comment (a-panayotopoulos) [inline src/red/java/com/lambdaherding/edi/red/ch02/Calculator.java:46]: Hmm, well, that would seem to be the latter value divided by the former value, which sounds odd to me. But ehhhh.

File: src/red/java/com/lambdaherding/edi/red/ch02/Calculator.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+		pan.add( inputButton( "4" ) );
+		pan.add( inputButton( "5" ) );
+		pan.add( inputButton( "6" ) );
+		pan.add( binaryActionButton( "*", (a, b) -> a * b ) );
+		pan.add( unaryActionButton( "1/x", x -> 1 / x ) );
+
+		pan.add( inputButton( "1" ) );
+		pan.add( inputButton( "2" ) );
+		pan.add( inputButton( "3" ) );
+		pan.add( binaryActionButton( "-", (a, b) -> -1 * a + b ) );
```
Comment (a-panayotopoulos) [inline src/red/java/com/lambdaherding/edi/red/ch02/Calculator.java:58]: And similarly here.

**Review by a-panayotopoulos — APPROVED** [review]
Pretty straightforward. Not sure about your operator order for divide/subtract, but perhaps it was intentional?

File: src/red/java/com/lambdaherding/edi/red/ch02/Calculator.java (hunk 1 above)
Comment (utopia-2) [reply src/red/java/com/lambdaherding/edi/red/ch02/Calculator.java:46]: Is that not what was happening in the original?

Comment (a-panayotopoulos) [reply src/red/java/com/lambdaherding/edi/red/ch02/Calculator.java:46]: Ah, was comparing my solution, and not realising that I was flipping the operands at a different point. That explains it.

**Pull request merged**